apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/ISpannerTransaction.cs

```diff
@@ -28,6 +28,9 @@ internal interface ISpannerTransaction
         // do if they need to return an int result type.
         Task<long> ExecuteDmlAsync(ExecuteSqlRequest request, CancellationToken cancellationToken, int timeoutSeconds);
 
+        // Note: this returns Task<ReliableStreamReader> to reflect requested affetced rows values from Spanner faithfully.
+        Task<ReliableStreamReader> ExecuteDmlReturnAsync(ExecuteSqlRequest request, CancellationToken cancellationToken, int timeoutSeconds);
```

Comment:
I'd rename this to `ExecuteDmlReader` or similar - because it will only ever be used with DML, it will always return a reader - but we don't actually know whether there's a "Then Return" clause.